hpedockerplugin/file_manager.py

```diff
@@ -0,0 +1,479 @@
+import base64
+import copy
+import json
+import socket
+import string
+import sh
+import six
+from Crypto.Cipher import AES
+from threading import Thread
+
+from oslo_log import log as logging
+from oslo_utils import netutils
+
+from hpedockerplugin.cmd import cmd_createshare
+from hpedockerplugin.cmd import cmd_generate_fpg_vfs_names
+from hpedockerplugin.cmd import cmd_setquota
+from hpedockerplugin.cmd import cmd_deleteshare
+
+import hpedockerplugin.exception as exception
+import hpedockerplugin.fileutil as fileutil
+import hpedockerplugin.hpe.array_connection_params as acp
+from hpedockerplugin.i18n import _
+from hpedockerplugin.hpe import hpe_3par_mediator
+from hpedockerplugin import synchronization
+
+LOG = logging.getLogger(__name__)
+
+
+class FileManager(object):
+    def __init__(self, host_config, hpepluginconfig, etcd_util,
+                 fp_etcd_client, node_id, backend_name='DEFAULT'):
+        self._host_config = host_config
+        self._hpepluginconfig = hpepluginconfig
+        self._my_ip = netutils.get_my_ipv4()
+
+        self._etcd = etcd_util
+        self._fp_etcd_client = fp_etcd_client
+        self._node_id = node_id
+        self._backend = backend_name
+
+        self._initialize_configuration()
+
+        self._decrypt_password(self.src_bkend_config,
+                               self.tgt_bkend_config,
+                               backend_name)
+
+        # TODO: When multiple backends come into picture, consider
+        # lazy initialization of individual driver
+        try:
+            LOG.info("Initializing 3PAR driver...")
+            self._primary_driver = self._initialize_driver(
+                host_config, self.src_bkend_config, self.tgt_bkend_config)
+
+            self._hpeplugin_driver = self._primary_driver
+            LOG.info("Initialized 3PAR driver!")
+        except Exception as ex:
+            msg = "Failed to initialize 3PAR driver for array: %s!" \
+                  "Exception: %s"\
+                  % (self.src_bkend_config.hpe3par_api_url,
+                     six.text_type(ex))
+            LOG.info(msg)
+            raise exception.HPEPluginStartPluginException(
+                reason=msg)
+
+        # If replication enabled, then initialize secondary driver
+        if self.tgt_bkend_config:
```

Comment:
@imran-ansari This is unused code please remove it

Reply:
Now that replication is not required, we can remove it. Will take care of it.